File: python/boj/deque/10866.py
import sys
from collections import deque

case = int(sys.stdin.readline())

queue = deque()

for _ in range(case):
    x = sys.stdin.readline().split()
    if x[0] == 'push_front':
        queue.insert(0,x[1])
    elif x[0] == 'push_back':
        queue.append(x[1])
    elif x[0] == 'front':
        if len(queue) != 0:
            print(queue[0])
        else:
            print(-1)
    elif x[0] == 'pop_front':
        if len(queue) != 0:
            print(queue.popleft())
        else:
            print(-1)
    elif x[0] == 'pop_back':
        if len(queue) != 0:
            print(queue.pop())
        else:
            print(-1)
    elif x[0] == 'empty':
        if len(queue) == 0:
            print(1)
        else:
            print(0)
    elif x[0] == 'back':
        if len(queue) != 0:
            print(queue[-1])
        else:
            print(-1)
    elif x[0] == 'size':
        print(len(queue))

File: cpp/queue/10866.cpp
#include <iostream>
#include <deque>
#include <string>

using namespace std;

int main(){
    cin.sync_with_stdio(0);
    cin.tie(0);


    deque<int> q;
    string a;
    int b;

    int x;
    cin >> x;

    for(int i=0;i<x;i++){
        cin >> a;
        if(a=="push_back"){
            cin >> b;
            q.push_back(b);
        }else if(a=="push_front"){
            cin >> b;
            q.push_front(b);
        }else if(a=="front"){
            if(q.empty()){
                cout << -1 << '\n';
            }else{
                cout << q.front() << '\n';
            }
        }else if(a=="pop_front"){
            if(!q.empty()){
                cout << q.front() << '\n';
                q.pop_front();
            }else{
                cout << -1 << '\n';
            }
        }else if(a=="pop_back"){
            if(!q.empty()){
                cout << q.back() << '\n';
                q.pop_back();
            }else{
                cout << -1 << '\n';
            }
        }else if(a=="size"){
            cout << q.size() << '\n';
        }else if(a=="empty"){
            if(q.empty()){
                cout << 1 << '\n';
            }else{
                cout << 0 << '\n';
            }
        }else if(a=="back"){
            if(!q.empty()){
                cout << q.back() << '\n';
            }else{
                cout << -1 << '\n';
            }
        }
    }
    return 0;
}